packages/in_app_purchase/pubspec.yaml

```diff
@@ -36,5 +32,5 @@ flutter:
         pluginClass: InAppPurchasePlugin
 
 environment:
-  sdk: ">=2.3.0 <3.0.0"
+  sdk: ">=2.12.0 <3.0.0"
```

Comment:
Based on the latest guidance from https://dart.dev/null-safety/migration-guide#sdk-constraints I believe this should be `2.12.0-259.9.beta`. (Technically it says you should use whatever you tested with, but I think for maximal compatibility using the oldest nullability-stable version is the right thing for this repo right now.)

Reply:
SG. Do we also update the example app to have the same version constraint? (>= 2.12.0-259.9.beta)

Reply:
We may as well. I don't see it mattering if we forget for some examples, but it wouldn't hurt to be consistent.